utils/swift_build_support/swift_build_support/products/indexstoredb.py

```diff
@@ -85,7 +85,7 @@ def get_dependencies(cls):
 
 
 def run_build_script_helper(action, host_target, product, args,
-                            sanitize_all=False, clean=True):
+                            sanitize_all=False, clean=False):
```

Comment:
> Won’t this break the indexstore-db build, which would now pass --no-clean, which isn’t accepted by the build-script-helper.py?

It might, haven't tried that recently.

> My proposal would be to just remove the cleaning step from SourceKit-LSP’s build-script-helper.py, never pass --no-clean and remove the clean parameter here.

That would defeat my purpose of optionally being able to clean sourcekit-lsp by manually overriding these flags when wanted.

Do you not want that cleaning step at all in the sourcekit-lsp `build-script-helper.py`? I would prefer to resurrect it in some way, but disabled in these scripts by default.

Let me know which approach you want and I will change these two pulls to do that.

Reply:
I would prefer to not have the `clean` step as it prevents us from re-using build artifacts from eg. the `build` phase in the `test` phase, which increases build time. If cleaning is required between builds, I think it should be done on a higher level.

Reply:
I was asking a different question, I think. I see two options here:

1. Remove all code related to cleaning `sourcekit-lsp` builds.
2. Bring back the cleaning code in the `sourcekit-lsp` repo, but disable using it for all build actions by default.

I have now implemented 2. here and in swiftlang/sourcekit-lsp#1558, as it's the route I prefer. Let me know what you think.

Reply:
I would prefer option (1). Other build scripts (eg. indexstore-db, swift-syntax) also don’t have a clean option.

Reply:
I think most have a way to clean the build as part of the `build-script` and they're usually enabled by default, whether in [their build-script-helper script](https://github.com/swiftlang/swift-driver/blob/22b351b913ad10ca53c35942584196b8ba38d6bf/Utilities/build-script-helper.py#L636) or [the bash portion of the overall `build-script`](https://github.com/swiftlang/swift/blob/f3acbb079fa766eb357b16b5925cb4bd729b1e6f/utils/build-script-impl#L2475). I know because I added a lot of the flags to disable them when wanted. 😉

Reply:
I still don’t think we should need it but I don’t feel super strongly about it. If you have strong opinions on it being useful, feel free to add it. 

But still, is anything ever passing `True` to `indexstoredb.run_build_script_helper`?

Reply:
> I still don’t think we should need it but I don’t feel super strongly about it. If you have strong opinions on it being useful, feel free to add it.

I'd like to have it, in case I need to turn on cleaning for my local builds at some point, as I did recently.

> But still, is anything ever passing True to indexstoredb.run_build_script_helper?

No this cleaning flag is set to false in this pull. The idea is just to have the flag available and working, so I can set it to true manually when I need it for my local builds.

If you're okay with that, please review and we can get these two pulls in.